### PR TITLE
Integrate react with registration editing

### PR DESCRIPTION
**Pull request by tomasr8**
Basic registration editing works for both normal users and management

The file field currently does not work properly (i.e. you have to upload a file everytime)
 since the backend returns the filename instead of thee uuid.

### CODE REVIEW

File: indico/modules/events/registration/controllers/__init__.py [hunk 1]
```diff
@@ -40,11 +42,13 @@ def _process_POST(self):
         if self.management:
             session['registration_notify_user_default'] = notify_user
         modify_registration(self.registration, form_data, management=self.management, notify_user=notify_user)
-        return redirect(self.success_url)
+        return jsonify({'redirect': self.success_url})
 
     def _process_GET(self):
+        form_data = get_flat_section_submission_data(self.regform)
+        # Filter out registration data which are not in the regform anymore
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/__init__.py:49]: are you sure this is the old behavior? don't we still show those fields if they were used in the registration? i know for sure that we do this with removed options of choice fields...

File: indico/modules/events/registration/controllers/__init__.py [hunk 2]
```diff
@@ -40,21 +46,32 @@ def _process_POST(self):
...
 
         registration_metadata = {
             'paid': self.registration.is_paid,
             'manager': self.management
         }
 
+        user_data = {t.name: getattr(session.user, t.name, None) if session.user else '' for t in PersonalDataType}
+        user_data['title'] = get_title_uuid(self.regform, user_data['title'])
         return self.view_class.render_template(self.template_file, self.event,
                                                sections=section_data, regform=self.regform,
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/__init__.py:67]: maybe you can rename this to `angular_sections` like i did in the other places as well so it's clear it's something to remove eventually?

Comment (tomasr8) [reply indico/modules/events/registration/controllers/__init__.py:67]: I'm actually using the user data to filter out the `initialValues` in `RegistrationFormSubmission`. The reason for that is that we have `unknown = RAISE` in the dynamically generated MM schema. If we eventually change that, we can also get rid of the `initialValues` filtering and also the user data

Comment (tomasr8) [reply indico/modules/events/registration/controllers/__init__.py:65]: I looked at it again and noticed that we don't use the `user_data` at all in editing (neither angular, nor react) so I removed it

File: indico/modules/events/registration/client/js/form_setup/index.jsx [hunk 3]
```diff
@@ -25,6 +25,7 @@ export default function setupRegformSetup(root) {
       eventStartDate,
       eventEndDate,
       currency,
+      paid: false,
```
Comment (ThiefMaster) [inline indico/modules/events/registration/client/js/form_setup/index.jsx:28]: I'd use `|| false` in the selector instead (like for `getManagement`)

**Pull request merged**